port/unix/omrsysinfo.c

```diff
@@ -7557,3 +7557,86 @@ omrsysinfo_get_number_context_switches(struct OMRPortLibrary *portLibrary, uint6
 	return OMRPORT_ERROR_SYSINFO_NOT_SUPPORTED;
 #endif /* defined(LINUX) */
 }
+
+/*
+ * Get the process ID and commandline for each process.
+ * @param[in] portLibrary The port library.
+ * @param[in] callback The function to be invoked for each process with the process ID and command info.
+ * @param[in] userData Data passed to the callback.
+ * @return 0 on success, or the first non-zero value returned by the callback.
+ */
+uintptr_t
+omrsysinfo_get_processes(struct OMRPortLibrary *portLibrary, OMRProcessInfoCallback callback, void *userData)
+{
+#if defined(LINUX)
+	struct dirent *entry = NULL;
+	char path[PATH_MAX];
+	char command[PATH_MAX];
+	uintptr_t callback_result = 0;
+	intptr_t bytes_read = 0;
+	intptr_t i = 0;
+	uintptr_t pid;
+	uintptr_t file = 0;
+	DIR *dir = opendir("/proc");
+	if (NULL == dir) {
+		int32_t rc = findError(errno);
+		portLibrary->error_set_last_error(portLibrary, errno, rc);
+		Trc_PRT_sysinfo_get_processes_failedOpeningProcFS(rc); 
+		return (uintptr_t)(intptr_t)rc;
+	}
+	for (;;) {
+		entry = readdir(dir);
+		if (NULL == entry) {
+			break;
+		}
+		/* Skip entries with no name. */
+		if ('\0' == entry->d_name[0]) {
+			continue;
+		}
+		/* Convert name to pid, skipping non-numeric entries. */
+		char *end = NULL;
+		pid = (uintptr_t)strtoull(entry->d_name, &end, 10);
+		if ('\0' != *end) {
+			continue;
+		}
+		command[0] = '\0';
+		/* Try reading /proc/[pid]/cmdline. */
+		portLibrary->str_printf(portLibrary, path, sizeof(path), "/proc/%s/cmdline", entry->d_name);
+		file = portLibrary->file_open(portLibrary, path, EsOpenRead, 0);
+		if (0 != file) {
+			bytes_read = portLibrary->file_read(portLibrary, file, command, sizeof(command) - 1);
```

Comment:
This may truncate commands which can be much longer than `PATH_MAX` (think megabytes).